### ISSUE TITLE
Fix pgModeler ARM build

### ISSUE THREAD

**Issue by Andrew0Hill** [Andrew0Hill/NURS6293_environment#21]
QT is failing to build pgModeler via GitHub Actions, but build works locally.

```
#33 [linux/arm64 build-common  6/19] RUN cd ./pgmodeler &&     chmod +x ./build_pgmodeler.sh &&     ./build_pgmodeler.sh &&     rm ./build_pgmodeler.sh
#33 0.421 Cannot find file: CONFIG.
#33 0.595 /usr/lib/aarch64-linux-gnu/qt6/mkspecs/features/toolchain.prf:76: Variable QMAKE_CXX.COMPILER_MACROS is not defined.
#33 0.604 Project ERROR: failed to parse default search paths from compiler output
#33 0.763 make: *** No targets specified and no makefile found.  Stop.
#33 ERROR: process "/bin/sh -c cd ./pgmodeler &&     chmod +x ./build_pgmodeler.sh &&     ./build_pgmodeler.sh &&     rm ./build_pgmodeler.sh" did not complete successfully: exit code: 2

#24 [linux/amd64 build-common  6/19] RUN cd ./pgmodeler &&     chmod +x ./build_pgmodeler.sh &&     ./build_pgmodeler.sh &&     rm ./build_pgmodeler.sh
#24 CANCELED
```

**Comment by Andrew0Hill** [comment]
Returning to this issue, this is unfortunately not an easy fix. 

This issue is a bug in qmake/QT when running under QEMU. See (https://bugreports.qt.io/browse/QTBUG-98951, https://bugreports.qt.io/browse/QTBUG-101677, https://wiki.qt.io/Qt_6.4_Known_Issues).

Workaround is to either compile QT from source with the `-no-feature-forkfd_pidfd` flag, or to compile on an ARM native platform (currently GitHub Actions supports M1 Mac runners, which could work).

**Comment by Andrew0Hill** [comment]
I don't think any significant QT capabilities are lost with that flag.

My concern is mostly that the flag needs to be supplied when compiling _QT/qmake_ from source, not the pgModeler application (which runs  _qmake_ to generate a makefile).

Our current Dockerfile just installs the pre-compiled QT binary packages for AMD/ARM64, meaning we we need to modify the Dockerfile to compile QT from source (with the `no-feature-forkfd_pidfd` flag) for the ARM64 build. 

From prior experience, I think that compiling QT would likely take too long for a GitHub Actions script, so we would need to compile this separately and pull it in as a binary artifact for the ARM build. This is doable but might take some time to configure. 

I'm going to test out the Mac M1 runner idea first since that just requires a simple workflow file change (and may very well work since all of our previous images have been built on my M1 Mac with no issues)

**Comment by mgkahn** [comment]
Got it. I did not realize that we pull in a pre-compiled version of QT rather than compile from source. Are you thinking that you could use the Mac M1 runner to do both X64 and ARM64? Or use two different runners in one build -- one for X64 and a Mac M1 for ARM64? This is all Docker territory beyond my experience with Docker.

**Comment by Andrew0Hill** [comment]
Well my plan was to just run everything on the Mac M1 runner (building ARM64 and AMD64 images) but I just tried and evidently the Mac M1 runners do not support docker.

I'll try to just compile QT from source for ARM64 and pull it in as a binary.

**Comment by Andrew0Hill** [comment]
Fixed in v0.1.2, we use separately built binaries to avoid this issue altogether (bonus is faster image build times since we aren't compiling pgModeler).
